pkg/controller/storagecluster/reconcile.go

```diff
@@ -409,3 +466,85 @@ func newCephDaemonResources(custom map[string]corev1.ResourceRequirements) map[s
 
 	return resources
 }
+
+// newStorageClassDeviceSets converts a list of StorageDeviceSets into a list of Rook StorageClassDeviceSets
+func newStorageClassDeviceSets(storageDeviceSets []ocsv1.StorageDeviceSet, topologyMap *ocsv1.NodeTopologyMap) []rook.StorageClassDeviceSet {
+	var storageClassDeviceSets []rook.StorageClassDeviceSet
+
+	for _, ds := range storageDeviceSets {
+		resources := ds.Resources
+		if resources.Requests == nil && resources.Limits == nil {
+			resources = defaults.DaemonResources["osd"]
+		}
+
+		topologyKey := ds.TopologyKey
+		topologyKeyValues := []string{}
+		noPlacement := ds.Placement.NodeAffinity == nil && ds.Placement.PodAffinity == nil && ds.Placement.PodAntiAffinity == nil
+
+		if noPlacement {
+			if topologyKey == "" {
+				topologyKey = "zone"
+			}
+			if topologyMap != nil {
+				topologyKey, topologyKeyValues = topologyMap.GetKeyValues(topologyKey)
+			}
+		}
+
+		replica := ds.Replica
+		if replica == 0 {
+			replica = defaults.DeviceSetReplica
+		}
+
+		// This is a temporary hack in place due to limitations in
+		// the current implementation of the OCP console. The console
+		// is hardcoded to create a StorageCluster with a Count of 3,
+		// as made sense for the previous behavior, but it cannot be
+		// updated until the next z-stream release of OCP 4.2. This
+		// workaround is to enable the new behavior while the console
+		// is waiting to be updated.
+		// TODO: Remove this behavior when OCP console is updated
+		count := ds.Count / 3
```

Comment:
This workaround is my only remaining reservation about this PR. I'm wondering if there's a safer way for us to express this workaround that won't inadvertently conflict with future configs. 

For example, as it is here if we set ds.Replica = 3 and ds.Count = 3, this work around would only give us 3 osds instead of 6.

What if we just checked to see if ds.Replica == 0 as the condition for this workaround. If Replica = 0, then attempt to put each "Count" on an individtual StorageClassDeviceSet

It wouldn't be a workaround anymore at that point. If you want the behavior of 1 replica with multiple OSDs, then you'll need to use Replica and Count.  If you only specify Count, then you're getting a StorageClassDeviceSet for each Count by default. 

Does that make sense?

Reply:
It does. I've moved the workaround into the `if replica == 0` stanza.